### PR TITLE
Env Variables created for each of the ports in addition to env variables...

### LINKED ISSUE #9900: Environment variables for published ports changed when using --link
I have multiple ports published from command line, ie.

   docker run --name backend -p 7101:7101 -p 7102:7102 -p 7103:7103 ....
   docker run --name frontedn --link backend:backend ...

There used to be environment variable available for each port, ie.

    BACKEND_PORT_7101_TCP_PORT=7101

But now there's only 

    BACKEND_PORT_7101_TCP_PORT_END=7103
    BACKEND_PORT_7101_TCP_PORT_START=7101

I think that https://github.com/docker/docker/commit/fd774a818c7d8942922b4f74eabd2a4e14094e1a changed the behaviour, and it requires at least three ports to be defined

This can be seen with

    docker run -d --name backend -p 7101:7101 -p 7102:7102 -p 7103:7103 busybox /bin/sh -c "while true; do sleep 5; done"

and then

    docker run -t -i --rm --name frontend --link backend:backend busybox /bin/sh -c set

### PR DESCRIPTION
**Pull request by brahmaroutu**
... for  port ranges, regression from #1834

Closes #9900
Signed-off-by: Srini Brahmaroutu srbrahma@us.ibm.com

### CODE REVIEW

**Comment by jfrazelle** [comment]
Can you please sign your commits following these rules:

https://github.com/docker/docker/blob/master/CONTRIBUTING.md#sign-your-work

The easiest way to do this is to amend the last commit:

`git commit --amend -s --no-edit && git push -f`

File: links/links.go [hunk 1]
```diff
@@ -91,8 +91,12 @@ func (l *Link) ToEnv() []string {
 
 			i = j + 1
 			continue
+		} else {
+			i++
 		}
-
+	}
+	for i := 0; i < len(l.Ports); {
```
Comment (LK4D4) [inline links/links.go]: Why not just `for _, p := range l.Ports`?

Comment (brahmaroutu) [inline links/links.go]: Yes no reason now, just a copy of old code. I changes it now.

**Comment by crosbymichael** [comment]
LGTM

**Comment by crosbymichael** [comment]
We should probably add a few integration tests to enforce the environment variables for a container or it maybe pretty easy to write a unit test for this method.

**Comment by LK4D4** [comment]
Yup, +1 to tests.

**Comment by brahmaroutu** [comment]
@crosbymichael @LK4D4 Thanks for the review. I will add tests shortly. One scenario that I am not very convinced about is that env for range may show differently, for example 
docker run --name backend -p 7101:7101 -p 7102:7102 -p 7103-7105:7103-7105
<we get normal ENV variables for all 5 ports with this fix but the range env variables will look like below since we detect the continuity in allocation>  
BACKEND_PORT_7101_TCP_PORT_END=7105
BACKEND_PORT_7101_TCP_PORT_START=7101

**Comment by brahmaroutu** [comment]
Please see that I added a unit test for check all env variables.

**Comment by icecrime** [comment]
LGTM

**Comment by icecrime** [comment]
@LK4D4 Anything else to add to this or can we merge?

**Comment by crosbymichael** [comment]
Thanks for the tests!

**Pull request merged**